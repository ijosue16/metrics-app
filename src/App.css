.App {
  text-align: center;
  height: 100%;
  background-color: #cf4278;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='48' height='48' viewBox='0 0 48 48'%3E%3Cg fill='%23e04683' fill-opacity='0.4'%3E%3Cpath d='M12 0h18v6h6v6h6v18h-6v6h-6v6H12v-6H6v-6H0V12h6V6h6V0zm12 6h-6v6h-6v6H6v6h6v6h6v6h6v-6h6v-6h6v-6h-6v-6h-6V6zm-6 12h6v6h-6v-6zm24 24h6v6h-6v-6z'%3E%3C/path%3E%3C/g%3E%3C/svg%3E");
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
