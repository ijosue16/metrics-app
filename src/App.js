import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import './App.css';
import Details from './components/details/details';
import Homepage from './components/homePage';
import Navbar from './components/navbar/navbar';

function App() {
  return (
    <div className="App">
      <Router>
        <Navbar />

        <Routes>
          <Route path="/" element={<Homepage />} />
          <Route path="/details" element={<Details />} />
        </Routes>
      </Router>
    </div>
  );
}

export default App;
