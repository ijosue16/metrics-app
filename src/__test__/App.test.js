import '@testing-library/jest-dom';

import { Provider } from 'react-redux';

import { render } from '@testing-library/react';

import {
  pickCity, Getweather,
} from '../redux/weather';
import store from '../redux/configureStore';
import App from '../App';

describe('test pure functions', () => {
  test('App component to render', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('test the type of Getweather function', async () => {
    const data = await Getweather();
    expect(data).toBeInstanceOf(Function);
    expect(data.payload).not.toBeNull();
  });
  test('test of Pickcity function', () => {
    const data = pickCity('Paris');
    expect(data).toBeInstanceOf(Object);
    expect(data).not.toBeNull();
    expect(data.payload).toEqual('Paris');
  });
});
