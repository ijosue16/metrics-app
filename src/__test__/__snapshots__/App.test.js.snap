// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test pure functions App component to render 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="d-flex py-2 container-fluid gap-2 navigation-bar"
    >
      <div
        class="flex-fill d-flex justify-content-start align-items-center"
      >
        <a
          aria-current="page"
          class="text-start text-decoration-none active"
          href="/"
        >
          <svg
            class="text-white"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.224 1.553a.5.5 0 0 1 .223.67L6.56 8l2.888 5.776a.5.5 0 1 1-.894.448l-3-6a.5.5 0 0 1 0-.448l3-6a.5.5 0 0 1 .67-.223z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </div>
      <div
        class="flex-fill d-flex justify-content-center align-items-center"
      >
        <h5
          class="text-center m-0 text-white"
        >
          Meteo forecast
        </h5>
      </div>
      <div
        class="flex-fill d-flex icons gap-4 gap-sm-5 justify-content-sm-end justify-content-center align-items-center"
      >
        <svg
          class="text-white"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
          />
        </svg>
        <svg
          class="text-white"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416.3 256c0-21 13.1-38.9 31.7-46.1-4.9-20.5-13-39.7-23.7-57.1-6.4 2.8-13.2 4.3-20.1 4.3-12.6 0-25.2-4.8-34.9-14.4-14.9-14.9-18.2-36.8-10.2-55-17.3-10.7-36.6-18.8-57-23.7C295 82.5 277 95.7 256 95.7S217 82.5 209.9 64c-20.5 4.9-39.7 13-57.1 23.7 8.1 18.1 4.7 40.1-10.2 55-9.6 9.6-22.3 14.4-34.9 14.4-6.9 0-13.7-1.4-20.1-4.3C77 170.3 68.9 189.5 64 210c18.5 7.1 31.7 25 31.7 46.1 0 21-13.1 38.9-31.6 46.1 4.9 20.5 13 39.7 23.7 57.1 6.4-2.8 13.2-4.2 20-4.2 12.6 0 25.2 4.8 34.9 14.4 14.8 14.8 18.2 36.8 10.2 54.9 17.4 10.7 36.7 18.8 57.1 23.7 7.1-18.5 25-31.6 46-31.6s38.9 13.1 46 31.6c20.5-4.9 39.7-13 57.1-23.7-8-18.1-4.6-40 10.2-54.9 9.6-9.6 22.2-14.4 34.9-14.4 6.8 0 13.7 1.4 20 4.2 10.7-17.4 18.8-36.7 23.7-57.1-18.4-7.2-31.6-25.1-31.6-46.2zm-159.4 79.9c-44.3 0-80-35.9-80-80s35.7-80 80-80 80 35.9 80 80-35.7 80-80 80z"
          />
        </svg>
      </div>
    </nav>
    <div
      class="container-fluid d-flex justify-content-center m-0 px-0 flex-column "
    >
      <div
        class="m-0 p-0 search d-flex align-items-center justify-content-between px-2"
      >
        <div
          class=" d-flex  align-items-center justify-content-start glb"
        >
          <span
            class="d-flex "
          />
        </div>
        <input
          class="p-2 m-0 search-bar d-flex justify-content-center rounded border-0 text-white "
          placeholder="Search a city"
          type="text"
        />
        <div
          class=" d-flex flex-column    align-items-end justify-content-end"
        >
          <h5
            class="flex-wrap global-dtl"
          >
            METEO AP PRICISE FORECAST
          </h5>
        </div>
      </div>
      <div
        class="d-flex justify-content-center "
      >
        <div
          class="container-fluid text-center p-0"
        >
          <h6
            class="py-1  container-fluid m-0 d-flex justify-content-start sub-head  global-dtl"
          >
             select city
          </h6>
          <div
            class="row row-cols-2 row-cols-md-4 container-fluid m-0 p-0 city-container "
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
